### DISCUSSION TITLE
[@types/react] and [@types/react-dom] React v18 mismatch in useReducer typedoc

### DISCUSSION THREAD

**Discussion by bencodesall** [DefinitelyTyped/DefinitelyTyped#67036] [Issues with a @types package]
The type-doc in the provided typings are showing 16.8.2 versions instead of my installed v18.2.0 for library documentation in IDE. See SO post [here](https://stackoverflow.com/questions/77268119/how-do-i-get-the-correct-latest-version-of-react-quick-documentation-in-websto?noredirect=1#comment136225351_77268119)See screenshots below. Obviously the type documentations are not up to date, per the first screenshot. Funny thing is that the `useReducer` signatures don't match the error showing up in my ide. Third argument is supposed to be `?` optional.

<img width="917" alt="image" src="https://github.com/DefinitelyTyped/DefinitelyTyped/assets/47954098/5a88a55e-1b5c-4dfc-8ebd-2cf1a0ccc2ba">

<img width="1114" alt="image" src="https://github.com/DefinitelyTyped/DefinitelyTyped/assets/47954098/83eb6ffa-bbef-46bc-adc7-4efc7016b25c">

<img width="1307" alt="image" src="https://github.com/reactjs/react.dev/assets/47954098/807fe395-8490-4478-b7d6-1fe6b2b3cfa1">

<img width="1492" alt="image" src="https://github.com/reactjs/react.dev/assets/47954098/6df45816-556c-43e3-bf8b-0d892745a4a1">

[a comment by an automated account is not shown]

**Reply by bencodesall** [reply]
[@types/react] and [@types/react-dom] do not have the correct versions reflecting installed package at v18. v17 looks like the highest build as of this date.

**Reply by bencodesall** [reply]
My bigger question, out of pure ignorance, is how does the react library / DT and "doc tooling" (or inspection) arrive at what get's layered into the code-inspections in my IDE. At first glance this seems like a constantly moving target. ¯l_(ツ)_l¯

**Comment by Semigradsky** [comment]
Why you think that "are showing 16.8.2 versions instead of my installed v18.2.0"?

I think that `@version 16.8.0` means version when these hooks were introduced. Like `@since` in `@types/node`

**Comment by Semigradsky** [comment]
> Funny thing is that the useReducer signatures don't match the error showing up in my ide. Third argument is supposed to be ? optional.

But it told you that `Expected 2-3 arguments`. `useReducer` is overloaded function and `2 arguments` is a variant without third argument.

**Reply by bencodesall** [reply]
The IDE was indicating a problem not receiving the 3rd argument, which (correctly) the react documentation shows as optional. However the @types/react does not show as optional.

See message `invalid number of arguments: expects 3`
<img width="1041" alt="image" src="https://github.com/DefinitelyTyped/DefinitelyTyped/assets/47954098/998300e2-af73-46f2-bdb1-9e7a830fd830">

**Comment by bencodesall** [comment]
nvmd, probs more a tool ui design confusion on my side within the IDE itself. Sorry to make noise.
